components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -378,6 +464,14 @@ func (w *NotebookWebhook) Handle(ctx context.Context, req admission.Request) adm
 		if ServiceMeshIsEnabled(notebook.ObjectMeta) {
 			return admission.Denied(fmt.Sprintf("Cannot have both %s and %s set to true. Pick one.", AnnotationServiceMesh, AnnotationInjectOAuth))
 		}
+
+		// Validate Auth sidecar resource configuration first
+		_, err := parseAndValidateAuthSidecarResources(notebook)
+		if err != nil {
+			return admission.Denied(fmt.Sprintf("invalid OAuth proxy resource configuration: %v", err))
+		}
```

Comment:
is this needed ?
seems like the injectoauthproxy seems to be including it ?

Reply:
Yeah, this was added just to return the `admission.Denied` instead of an InternalServerError that would be returned from the InjectOAuthProxy function couple of lines below. I admit that it's not ideal as this is called twice. I didn't like much other options to avoid this double call (e.g. provide `config` value to the InjectOAuthProxy, return various error types from the InjectOAuthProxy as I felt both are a bit of overkill). I can amend, though. Do you have any preference?